app.js

```diff
@@ -1,45 +1,73 @@
 const express = require("express");
 const fs = require("fs");
+const bodyParser = require("body-parser")
 
 const app = express();
 
 const port = 5000;
 
-app.get('/', (req, res) => {
-	res.send('Greeting from server');
+let products = [];
+
+fs.readFile('./db/products.json', 'utf-8', (err, data) => {
+	if (err) throw err;
+	products = JSON.parse(data);
 });
 
-app.get('/data', (req, res) => {
-	res.send('Here is some data');
+const writeToFile = (list, res) => {
+	fs.writeFile('./db/products.json', JSON.stringify(list), err => {
+		if (err) throw err;
+		res.send(list);
+	});
+};
+
+app.use(bodyParser.urlencoded({extended: false}));
+
+app.use(bodyParser.json());
+
+app.get('/products', (req, res) => {
+	res.send(products);
 });
 
-app.listen(port, () => {
-	console.log(`Server is listening on port ${port}`);
-})
+app.get('/products/:id', (req, res) => {
+	const { id } = req.params;
 
-let text = '';
-for (let i = 1; i <= 10; i++) {
-	text += i + '\n';
-}
+	const product = products.filter((product) => product.id === id);
 
-fs.writeFile('./text/file.txt', text, err => {
-	if (err) throw err;
-	console.log('File successfully written');
+	res.send(product);
 });
 
-fs.readFile('./text/file.txt', 'utf-8', (err, data) => {
-	if (err) throw err;
+app.post('/products', (req, res) => {
+	const { product } = req.body;
 
-	const arr = data.split('\n');
-	let randomText = arr[Math.floor(Math.random()*(arr.length - 1))];
+	products.push(product);
 
-	fs.appendFile('./text/file.txt', randomText, err => {
-		if (err) throw err;
-	});
+	writeToFile(products, res);
+});
 
-	console.log('Random text:', randomText)
+app.put('/products/:id', (req, res) => {
+	const { id } = req.params;
+	const { product } = req.body;
 
-	fs.unlink('./text/file.txt', err => {
-		if (err) throw err;
-	});
-});
\ No newline at end of file
+	products = products.map((existingProduct) => (
+		existingProduct.id === id ? product : existingProduct
+	));
+	writeToFile(products, res);
+});
+
+app.delete('/products/:id', (req, res) => {
+	const { id } = req.params;
+
+	products = products.filter((product) => product.id !== id);
```

Comment:
Better to use findIndex to find the index of necessary element and then delete that element with splice

Reply:
Okay